### ISSUE TITLE
My application cannot work correctly with the landmarks.

### ISSUE THREAD

**Issue by hady13** [googlesamples/android-vision#217]
I already referred this issue.
https://github.com/googlesamples/android-vision/issues/197

But I'm trying to draw landmark in real time video, not bitmap, so I have wrote like this.

![1](https://cloud.githubusercontent.com/assets/11660797/25310154/ff3a3dfa-2818-11e7-9bf8-4009bac81ca1.JPG)
![2](https://cloud.githubusercontent.com/assets/11660797/25310155/02e60394-2819-11e7-84a1-6767c522f35f.JPG)

But I still have wrong positions... From where do landmarks detect the position in this API?
Please help me.

**Comment by pm0733464** [comment]
I don't think that you have the scale right.  See this example:

https://github.com/googlesamples/android-vision/blob/master/visionSamples/photo-demo/app/src/main/java/com/google/android/gms/samples/vision/face/photo/FaceView.java#L73

**Comment by hady13** [comment]
Yes, as I said, I already referred that example. But that is for bitmap image, so I change the code like that... So How I chage this code..?

**Comment by JamesRArroyo** [comment]
@hady13 Did you ever figure out how to get the scale to properly place the landmarks in real time video? I am also having the same issue and have tried multiple things but no luck.

**Comment by JamesRArroyo** [comment]
Just got my landmarks to line up as they should by using the translate functions. The translate functions get the scale by subtracting the mOverlay width (or height) from the scaleFactor of the landmarks position. Posting this solution just in case someone else comes along. 
```
for (Landmark landmark : face.getLandmarks()) {
    int cx = (int) translateX(landmark.getPosition().x);
    int cy = (int) translateY(landmark.getPosition().y);
    canvas.drawCircle(cx, cy, 10, mBoxPaint);
}
```
